### PR TITLE
[spike] DCOS-2516: Migrate React.createClass

### PR DESCRIPTION
**Pull request by brandonc**
## Overview

Migrated React.createClass to ES6 classes, making it unnecessary to do so during the React 16 upgrade.

## Testing

I smoke tested all components that I migrated but the big risk is that I missed a constructor binding, so review for those if possible.

## Trade-offs

I could have simply added create-react-class module but that would have been one more dependency and this is more consistent anyway.

### CODE REVIEW

**Comment by GeorgiSTodorov** [comment]
File conflicts, because we merged https://github.com/dcos/dcos-ui/pull/3485 .

**Comment by GeorgiSTodorov** [comment]
Looks good. One small problem I found was in Nodes Grid View. The data was slow to load and meanwhile, it looked like this:
screenshot from 2019-01-09 15-50-08
I couldn't reproduce this problem on master.

**Comment by GeorgiSTodorov** [comment]
File conflicts.

**Comment by brandonc** [comment]
@GeorgiSTodorov I was unable to fix the issue you found yet... I will rebase when I do

File: src/js/components/Page.js [hunk 1]
```diff
@@ -72,10 +71,6 @@ class Page extends mixin(InternalStorageMixin, StoreMixin) {
     this.forceUpdate();
   }
 
-  onSidebarStoreWidthChange() {
```
Comment (nLight) [inline src/js/components/Page.js]: This is actually being used :)
When you subscribe to a store listener using the `StoreMixin` it comes with naming conventions :) Do you remember deleting more like this?

**Comment by nLight** [comment]
Rebased 🙌 Thanks for the spike!

File: plugins/nodes/src/js/components/NodesGridDials.js [hunk 2]
```diff
@@ -71,7 +65,7 @@ var NodesGridDials = createReactClass({
     percentage = Math.round(percentage);
 
     config.push({
-      colorIndex: colors.unused,
+      colorIndex: COLOR_INDEX_UNUSED,
```
Comment (nLight) [inline plugins/nodes/src/js/components/NodesGridDials.js]: Love use of `CONSTANTS`!

File: plugins/nodes/src/js/components/NodesGridDials.js [hunk 3]
```diff
@@ -10,29 +9,24 @@ import DialChart from "#SRC/js/components/charts/DialChart";
...
-  },
+class NodesGridDials extends React.Component {
+  constructor() {
+    super(...arguments);
 
-  contextTypes: {
-    router: routerShape
-  },
+    METHODS_TO_BIND.forEach(m => {
+      this[m] = this[m].bind(this);
```
Comment (nLight) [inline plugins/nodes/src/js/components/NodesGridDials.js]: 🙈 short `m`

File: plugins/services/src/js/components/FilterByFramework.js [hunk 4]
```diff
@@ -1,41 +1,31 @@
...
-  getDefaultProps() {
-    return {
-      byFrameworkFilter: defaultId,
-      frameworks: [],
-      totalHostsCount: 0,
-      handleFilterChange() {}
-    };
-  },
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
```
Comment (nLight) [inline plugins/services/src/js/components/FilterByFramework.js]: inconsistency 👮 
though 🙈

File: plugins/services/src/js/components/ServiceList.js [hunk 5]
```diff
@@ -120,19 +118,29 @@ const ServiceList = createReactClass({
         />
       </div>
     );
-  },
+  }
 
-  getContent() {
+  render() {
```
Comment (nLight) [inline plugins/services/src/js/components/ServiceList.js]: ❤️

File: src/js/components/Page.js [hunk 6]
```diff
@@ -164,14 +149,25 @@ const Page = createReactClass({
     return (
       <div className={classSet}>
         <MountService.Mount type="Page:TopBanner" />
-        {this.getPageHeader(title, navigation)}
+        {this.getPageHeader()}
```
Comment (nLight) [inline src/js/components/Page.js]: I deleted unused arguments here.

File: src/js/components/charts/BarChart.js [hunk 7]
```diff
@@ -1,392 +0,0 @@
-import classNames from "classnames";
```
Comment (nLight) [inline src/js/components/charts/BarChart.js]: Maybe also removing some styles
https://github.com/dcos/dcos-ui/blob/brandonc/spike-remove-react-create-class/src/styles/components/charts/styles.less#L154 - looks related

File: src/js/components/charts/Chart.js [hunk 8]
```diff
@@ -47,15 +42,12 @@ var Chart = createReactClass({
...
     global.addEventListener("resize", this.updateWidth);
-  },
+  }
 
   componentWillUnmount() {
+    this._mounted = false;
     global.removeEventListener("resize", this.updateWidth);
-  },
-
-  onSidebarStoreWidthChange() {
```
Comment (nLight) [inline src/js/components/charts/Chart.js]: we should put this one back ⚠️

File: src/js/components/charts/ResourceChart.js [hunk 9]
```diff
@@ -1,102 +0,0 @@
-import PropTypes from "prop-types";
```
Comment (nLight) [inline src/js/components/charts/ResourceChart.js]: could you please research dangling styles?

File: src/js/components/charts/Chart.js (hunk 8 above)
Comment (brandonc) [reply src/js/components/charts/Chart.js]: I couldn't find a caller at the time but now I see it's store listener magic

File: src/js/components/charts/ResourceChart.js (hunk 9 above)
Comment (brandonc) [reply src/js/components/charts/ResourceChart.js]: Looks good I think

**Comment by GeorgiSTodorov** [comment]
@brandonc , file conflicts.

**Comment by nLight** [comment]
rebasing

**Comment by nLight** [comment]
I'm merging it when it's green

**Comment by GeorgiSTodorov** [comment]
I will do some smoke testing just in case.

**Comment by GeorgiSTodorov** [comment]
From Jenkins:
```
Error: Cannot find module 'seek-bzip'

    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)

    at Function.Module._load (internal/modules/cjs/loader.js:507:25)

    at Module.require (internal/modules/cjs/loader.js:637:17)

    at require (internal/modules/cjs/helpers.js:22:18)

    at Object.<anonymous> (/jenkins/workspace/end_dcos-ui-oss-pipeline_PR-3494/node_modules/decompress/node_modules/decompress-tarbz2/index.js:5:18)

    at Module._compile (internal/modules/cjs/loader.js:689:30)

    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)

    at Module.load (internal/modules/cjs/loader.js:599:32)

    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)

    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
```

**Comment by nLight** [comment]
@GeorgiSTodorov fixed both 🙇

**Comment by brandonc** [comment]
🙏

**Comment by GeorgiSTodorov** [comment]
The integration tests failing here are failing locally as well. It looks like the table isn't shown for some reason. I think @mperrotti had and solved a similar problem in one of his PRs.
Screenshot from 2019-05-10 12-08-34

**Pull request closed without merging**